Sources/OpenAPIRuntime/Conversion/Converter+Common.swift

```diff
@@ -17,31 +17,53 @@ extension Converter {
 
     // MARK: Miscs
 
-    /// Validates that the Content-Type header field (if present)
-    /// is compatible with the provided content-type substring.
+    /// Returns the MIME type from the content-type header, if present.
+    /// - Parameter headerFields: The header fields to inspect for the content
+    /// type header.
+    /// - Returns: The content type value, or nil if not found or invalid.
+    public func extractContentTypeIfPresent(in headerFields: [HeaderField]) -> MIMEType? {
+        guard let rawValue = headerFields.firstValue(name: "content-type") else {
+            return nil
+        }
+        return MIMEType(rawValue)
+    }
+
+    /// Checks whether a concrete content type matches an expected content type.
     ///
-    /// Succeeds if no Content-Type header is found in the response headers.
+    /// The concrete content type can contain parameters, such as `charset`, but
+    /// they are ignored in the equality comparison.
     ///
+    /// The expected content type can contain wildcards, such as */* and text/*.
     /// - Parameters:
-    ///   - headerFields: Header fields to inspect for a content type.
-    ///   - substring: Expected content type.
-    /// - Throws: If the response's Content-Type value is not compatible with the provided substring.
-    public func validateContentTypeIfPresent(
-        in headerFields: [HeaderField],
-        substring: String
-    ) throws {
-        guard
-            let contentType = try getOptionalHeaderFieldAsText(
-                in: headerFields,
-                name: "content-type",
-                as: String.self
-            )
-        else {
-            return
+    ///   - received: The concrete content type to validate against the other.
+    ///   - expected: The expected content type, can contain wildcards.
+    /// - Returns: A Boolean value representing whether the concrete content
+    /// type matches the expected one.
+    public func isValidContentType(received: MIMEType?, expected: String) -> Bool {
```

Comment:
Yeah I don't love the name either, so suggestions welcome. I considered `isMatchingContentType`, `isCompatibleContentType`. It should be a sentence fragment that expects a Bool response, so should start with `is` or `has` or similar.

`validateContentTypeMatches` would make sense if it threw an error and returned Void, but maybe `doesContentTypeMatch`?

Reply:
I'm happy with anything that implies it's testing for loose equivalence, rather than the validity of the first parameter.

I think the sentence fragment argument would apply to a method or computed property. I don't think it _has_ to be here, which is essentially a pure function. I'd probably start with a verb, like `validateContentTypeMatches(actual:expected:)`. But I honestly don't mind at all. Any of your suggestions meet my criteria.

Reply:
Ok, went for `isMatchingContentType`, as that'll extend well for when we start supporting more wildcards as well.

Runtime: https://github.com/apple/swift-openapi-runtime/pull/29/commits/606aa0ee56db8cfee6f61108c4e75a5773571f52
Generator: https://github.com/apple/swift-openapi-generator/pull/146/commits/d31270ab01196f3836d0eca0d34885e3f47b3166